### PR TITLE
Format ObjC comment docs as Cocoadocs

### PR DESCRIPTION
**Pull request by jcanizales**
Part of #2853.

CC @nathanielmanistaatgoogle as alternative reviewer, as this is mostly a Python script.

### CODE REVIEW

File: src/objective-c/change-comments.py [hunk 1]
```diff
@@ -0,0 +1,102 @@
+#!/usr/bin/python
```
Comment (murgatroid99) [inline src/objective-c/change-comments.py]: Why are we committing this file? Do we expect to use it again?

Comment (nathanielmanistaatgoogle) [inline src/objective-c/change-comments.py]: This should probably be `#!/usr/bin/env python2.7`, as is done elsewhere in the codebase. (Do you see any place where we have `#!/usr/bin/python` in the codebase today?)

Comment (jcanizales) [inline src/objective-c/change-comments.py]: Michael - If someone ever sends a PR with //-style doc comments, it's best to have the script around.

File: src/objective-c/change-comments.py [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+import re
+import sys
+
+
+if len(sys.argv) < 2:
+  print("Please provide at least one source file name as argument.")
+  quit()
```
Comment (nathanielmanistaatgoogle) [inline src/objective-c/change-comments.py]: Is there supposed to be a `quit` function defined in this module?

File: src/objective-c/change-comments.py [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+for file_name in sys.argv[1:]:
+
+  print("Modifying format of {file} comments in place...".format(
+      file = file_name,
```
Comment (nathanielmanistaatgoogle) [inline src/objective-c/change-comments.py]: When passing an argument to a function by keyword, omit the spaces on either side of the equals sign.

File: src/objective-c/change-comments.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+  ))
+
+
+  # Input
+
+  lines = []
```
Comment (nathanielmanistaatgoogle) [inline src/objective-c/change-comments.py]: The value assigned to `lines` on this line cannot possibly be used before `lines` is reassigned to a different value. Isn't this dead code?

File: src/objective-c/change-comments.py [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+
+  # Output
+
+  output_lines = []
+
+  def write(line):
+    output_lines.append(line)
+
+  def flush_output():
+    with open(file_name, "w") as otuput_file:
```
Comment (nathanielmanistaatgoogle) [inline src/objective-c/change-comments.py]: Spelling error `otuput_file`.

File: src/objective-c/change-comments.py [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+  comment_regex = r'^(\s*)//\s(.*)$'
+
+  def is_comment(line):
+    return re.search(comment_regex, line)
+
+  def isnt_comment(line):
+    return not is_comment(line)
+
+  def next_line(predicate):
+    if not more_input_available():
```
Comment (nathanielmanistaatgoogle) [inline src/objective-c/change-comments.py]: `return more_input_available() and predicate(peek())`?

**Comment by nathanielmanistaatgoogle** [comment]
First pass through Python file complete.

File: src/objective-c/change-comments.py (hunk 6 above)
Comment (jcanizales) [inline src/objective-c/change-comments.py]: Nice. Done.

File: src/objective-c/change-comments.py (hunk 5 above)
Comment (jcanizales) [inline src/objective-c/change-comments.py]: Oops. Done.

File: src/objective-c/change-comments.py (hunk 4 above)
Comment (jcanizales) [inline src/objective-c/change-comments.py]: It is. Removed.

File: src/objective-c/change-comments.py (hunk 3 above)
Comment (jcanizales) [inline src/objective-c/change-comments.py]: Done.

File: src/objective-c/change-comments.py (hunk 2 above)
Comment (jcanizales) [inline src/objective-c/change-comments.py]: I just Googled how to exit a script in Python and saw it said `quit()` was preferable to `sys.exit()` as it wouldn't exit the interpreter. It doesn't seem to require any import and it does work.

File: src/objective-c/change-comments.py (hunk 1 above)
Comment (jcanizales) [inline src/objective-c/change-comments.py]: Nathaniel - Changed, plus added copyright notice and a short description of what the script does.

**Comment by jcanizales** [comment]
PTAL.

File: src/objective-c/change-comments.py (hunk 2 above)
Comment (nathanielmanistaatgoogle) [inline src/objective-c/change-comments.py]: Wow. Okay, I'd never before ever heard of `quit`, but it looks like it's documented at https://docs.python.org/2/library/constants.html#quit. Wow.

Comment (murgatroid99) [inline src/objective-c/change-comments.py]: I think I should point out the preceding line in that documentation:

> They are useful for the interactive interpreter shell and should not be used in programs.

**Comment by nathanielmanistaatgoogle** [comment]
Python portion of this pull request looks satisfactory to me. There's a lot that we could do to improve and polish it, but since it's a utility script and not actually part of production gRPC code I'm not sure that I'd want to put you through all the fuss.

File: src/objective-c/change-comments.py (hunk 2 above)
Comment (jcanizales) [inline src/objective-c/change-comments.py]: `sys.exit()` then?

Comment (murgatroid99) [inline src/objective-c/change-comments.py]: I think so

Comment (jcanizales) [inline src/objective-c/change-comments.py]: Done. Also tested just in case :)

**Pull request merged**